packages/core/notifications/core-notifications-browser-internal/src/toasts/global_toast_list.tsx

```diff
@@ -21,24 +22,27 @@ interface Props {
 
 interface State {
   toasts: Toast[];
+  idToToasts: Record<string, Toast[]>;
 }
 
 const convertToEui = (toast: Toast): EuiToast => ({
   ...toast,
-  title: typeof toast.title === 'function' ? <MountWrapper mount={toast.title} /> : toast.title,
-  text: typeof toast.text === 'function' ? <MountWrapper mount={toast.text} /> : toast.text,
+  title: toast.title instanceof Function ? <MountWrapper mount={toast.title} /> : toast.title,
+  text: toast.text instanceof Function ? <MountWrapper mount={toast.text} /> : toast.text,
```

Comment:
nit: why change to `instanceof` here?

Reply:
Typescript is not as happy with the `typeof` once I extended the title/text to allow for react components.
<img width="844" alt="Screenshot 2023-07-20 at 12 49 50" src="https://github.com/elastic/kibana/assets/4738868/bff38740-a06d-4d30-9424-c551c8d4bb9d">